src/Common/SensitiveDataMasker.cpp

```diff
@@ -74,21 +73,37 @@ class SensitiveDataMasker::MaskingRule
                 regexp_string_, regexp.error());
     }
 
-    uint64_t apply(std::string & data) const
+    uint64_t applyThrow(std::string & data) const
     {
         auto m = RE2::GlobalReplace(&data, regexp, replacement);
 
         if (throw_on_match && m > 0)
+        {
             throw Exception(ErrorCodes::LOGICAL_ERROR,
-                            "The rule {} was triggered on the log line {}",
-                            name, data);
+                "The rule {} was triggered on the log line {}",
+                name, data);
+        }
+
 
 #ifndef NDEBUG
         matches_count += m;
 #endif
         return m;
     }
 
+    uint64_t applyNoThrow(std::string & data) const
+    {
+        if (throw_on_match)
+            return 0;
+
+        auto m = RE2::GlobalReplace(&data, regexp, replacement);
```

Comment:
Seems we should still apply replace if `throw_on_match == 1` and return `m`?

Reply:
Yes, sure. I will try to address wrong usage of SensitiveDataMasker in a separate PR.